Autosurgeon code improvement and light overlay

CoffeeDragon16 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Fixes #68016. The organ loading behavior was put on a path with no behavior, this removes the now-defunct path. It also adds an overlay light to show when the autosurgeon has an organ inside of it. Cleans up a lot of the autosurgeon code as well. Adds combat logging for autosurgery implanting someone.

![dreamseeker_0hP2VNQ9iJ](https://user-images.githubusercontent.com/63162339/175824690-4ef639c8-4497-4cbe-9d22-79569d8a33f8.png)

## Why It's Good For The Game

Bugs bad. Autosurgeon light saves you an examine.

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
fix: you can now properly load fugitive autosurgeons
imageadd: light for autosurgeons to show when they have an organ in them
admin: autosurgery implanting someone now has combat logging
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

CoffeeDragon16 committed:
Does-the-PR

CoffeeDragon16 committed:
oop typo

CoffeeDragon16 committed:
another typo

CoffeeDragon16 committed:
var comment

CoffeeDragon16 committed:
clarifies proc name

CoffeeDragon16 committed:
improves custom organ naming

CoffeeDragon16 committed:
futureproofing

CoffeeDragon16 committed:
adds combat logging

CoffeeDragon16 committed:
fixes possible issue

CoffeeDragon16 committed:
used wrong proc

CoffeeDragon16 committed:
you know this worked the first time

CoffeeDragon16 committed:
moves more behavior to load_organ and use_autosurgeon, repaths health_hud

CoffeeDragon16 committed:
ack

CoffeeDragon16 committed:
updates comment, improves check

unit0016 commented on code/modules/surgery/organs/autosurgeon.dm:
```diff
@@ -103,24 +104,37 @@
+
+/obj/item/autosurgeon/syndicate/commsagent
+	desc = "A device that automatically - painfully - inserts an implant. It seems someone's specially \
+	modified this one to only insert... tongues. Horrifying."
```
Looks like this can use any type of organ now? I wanted to avoid doing that as it'd be a massive buff to anyone who raided the lavaland syndies (or now the space comms agent too, apparently)

CoffeeDragon16 replied:
It looks like the tongue organ whitelist never worked, that was just fluff. I added it in now though. The old comms agent implanter didn't even work in the first place, I think.

CoffeeDragon16 committed:
adds organ whitelist, moves /commsagent/ to it

CoffeeDragon16 committed:
adds organ whitelist

LemonInTheDark commented on code/modules/surgery/organs/autosurgeon.dm:
```diff
@@ -3,92 +3,108 @@
+	storedorgan = loaded_organ
+	loaded_organ.forceMove(src)
+	name = "[initial(name)] ([storedorgan.name])" //to tell you the organ type, like "suspicious autosurgeon (Reviver implant)"
+	add_overlay(loaded_overlay)
```
rather then managing the overlay like this, you should make it part of update_appearance
also should make it emissive, that'll look quite nice

CoffeeDragon16 committed:
fixes screwdriver_act, adds emissive & update_appearance()

CoffeeDragon16 committed:
space

Salex08 committed:
Crew starts with the right amount of breathing mask again (#68153)

removes unnecessary codepiece

tgstation-server committed:
Automatic changelog generation for PR #68153 [ci skip]

san7890 committed:
I'm Fucking Stupid - Actually Adds Eggs+Milk To IceBox (#68146)

WHY DOES THIS KEEP HAPPENING TO ME THERE'S LIKE FIFTEEN THOUSAND KITCHEN CABINETS WITH THE WORSE NAMES ON THIS PLANET EARTH AND I KEEP THINKING I HAVE THE RIGHT ONE BUT IT'S NEVER THE RIGHT FUCKING ONE

I also added a milk carton to the kitchen itself, as a reminder.

[x] i did test this time

tgstation-server committed:
Automatic changelog generation for PR #68146 [ci skip]

TheBoondock committed:
Pluoxium can now heal oxygen damage (#67990)

Pluoxium simply doesn't have much unique interaction beside requiring less partial pressure. This pr simply makes it more unique than being a better oxygen

tgstation-server committed:
Automatic changelog generation for PR #67990 [ci skip]

jlsnow301 committed:
Completely rewrites the orbit screen (#68054)

I set about refactoring Orbit.js when I just decided to scrap it and maybe add in some different functionality.

This PR:
- rewrites orbit from the ground up in typescript
- color coded list + scrollable + wider
- adds some styling and effects to gauge "threat": icons change, buttons change color
- fixes button text overflow
- fixes issue where scrolling hides the search header
- fixes similar antags being grouped separately (nukie and nukie leader, etc)

tgstation-server committed:
Automatic changelog generation for PR #68054 [ci skip]

MrMelbert committed:
Fixes some cases which references are used in trait sources, potentially causing hard deletes (#67974)


About The Pull Request

Fixes some cases in which actual references were used in trait sources instead of keys (or ref() keys).

This can cause some rare and difficult to find hard deletes.

Trait sources should be a string key relating to the source of it, not an actual reference to what added it. References within trait sources are never handled in Destroy(), because it's not expected behavior, meaning it can cause hanging references.

So, I went through with a regex to find some cases and replaced them.
I used the following and just picked through the few by hand to find erroneous ones.
ADD_TRAIT\(.+, .+, [a-z]+\)
REMOVE_TRAIT_TRAIT\(.+, .+, [a-z]+\)
Why It's Good For The Game

Less hard deletes, probably.
Changelog

cl Melbert
code: Some traits which mistakenly were sourced from a hard reference are no longer.
/cl

tgstation-server committed:
Automatic changelog generation for PR #67974 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

Commit:
Remove move and combat mode from the stat tab (#68150)

tgstation-server committed:
Automatic changelog generation for PR #68150 [ci skip]

Commit:
Replace a statbrowser verb call with send message (#68151)

tgstation-server committed:
Automatic changelog generation for PR #68151 [ci skip]

vincentiusvin committed:
Refactors canister damage calculation (#68142)

I think i managed to keep the old temp calculation one-to-one. Also added a currently unused ignore. Just to make the system more robust.

Temp calculation: Code nice :)
refactor: Generalized how the damage calculation is made in cans, pumps, and scrubbers. No significant gameplay changes

tgstation-server committed:
Automatic changelog generation for PR #68142 [ci skip]

Mooshimi committed:
Remove lights/signs in burn chambers (#68141)

Removes four lightbulbs from burn chambers on icebox and kilo ordinance, total.
Removes one mask sign from burn chamber of icebox atmos turbine.

They just burn when the chamber is lit so there doesn't seem to be a point to them.

tgstation-server committed:
Automatic changelog generation for PR #68141 [ci skip]

necromanceranne committed:
Adds windup autofire functionality. The future is now. (Later) (#67911)

Pulls the wind up autofire functionality from my energy weapon pr as a standalone code improvement.

tgstation-server committed:
Automatic changelog generation for PR #67911 [ci skip]

Salex08 committed:
Fixes a big oversight with steal guns objective (#68093)

tgstation-server committed:
Automatic changelog generation for PR #68093 [ci skip]

Fikou committed:
Revert "Makes supermatter dusting independent of the reference frame." (#66491)



Reverts #66395
Can we not have supermatter launching murderbones? This was merged with 0 comment or discussion from the maintainer, with a literal 1 word "Why it's good for the game section". The PR was opened literally like 3 minutes after a discord discussion on why doing that would be a fucking bad idea

cl
balance: The supermatter running into you is no longer just as deadly as you running into the supermatter.
/cl

tgstation-server committed:
Automatic changelog generation for PR #66491 [ci skip]

RandomGamer123 committed:
Add a scrubber and vent to a part of a hallway on Meta that didn't have them (#68165)

* Add a scrubber and vent to a part of a hallway on Meta that didn't have them

* Fix mapmerge error

tgstation-server committed:
Automatic changelog generation for PR #68165 [ci skip]

GuillaumePrata committed:
Balloonify and removes some `to_chat` spam for miners (#67816)


About The Pull Request

Playing as miner and paying attention to the radio is pure pain, so many of your job related actions create useless spam messages inside your chat/radio tab (unless you split them, which creates other issues) so this PR changes a few of them:

1 - PKAs won't send a useless * click * message to your chat window when it's not ready to fire. It already has a different and clear sound that does it's job.
2 - Guns won't send a message to your chat with "you shoot the gun". You clicked your mouse, you hear the sound, you see the projectile leave and you already get a message logged there with what you hit.
3 - Digging with a pickaxe/shovel/hand is a balloon alert and the message for when you finish digging a wall is removed as you just seen the wall of basalt disappear so you know what happened...
Why It's Good For The Game

As a miner most of your interaction with the station is over the radio, less useless spam on the chat tab will hopefully help make mining feel less like a single player game.
Changelog

cl Guillaume Prata
qol: Miners rejoice! A few of the mostly useless and spammy messages that get send to your chat were changed to balloon alerts or straight up removed. Hopefully, you will have an easier time listening to the radio chatter now.
/cl

tgstation-server committed:
Automatic changelog generation for PR #67816 [ci skip]

zzzmike committed:
Update quartermaster.dm with head of staff announcement (#68130)

Update quartermaster.dm

tgstation-server committed:
Automatic changelog generation for PR #68130 [ci skip]

zzzmike committed:
Updates jobmanagement.dm to blacklist QM (#68128)

tgstation-server committed:
Automatic changelog generation for PR #68128 [ci skip]

zzzmike committed:
Updates CrewManifest.js with QM as command (#68127)

tgstation-server committed:
Automatic changelog generation for PR #68127 [ci skip]

JohnFulpWillard committed:
Fixes Thief's preference icon (#68107)

* Fixes thief's prefs

* screenshot test

tgstation-server committed:
Automatic changelog generation for PR #68107 [ci skip]

necromanceranne committed:
Fixes some body part coverage of the pirate outfit and justice outfit, stops golems wearing an armored suit (#68129)

Fixes the coverage of the pirate costume and justice suit so that their armor can apply properly, prevents golems from wearing the armored version

tgstation-server committed:
Automatic changelog generation for PR #68129 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

orthography committed:
Changed a "lil pump" to a normal pump. (#68177)

Changes one of the portable air pumps in the ordinance lab on icebox from "lil pump" to a normal portable air pump to match the standard of every other map.

I hate fun. And give my life to San.

tgstation-server committed:
Automatic changelog generation for PR #68177 [ci skip]

Iamgoofball committed:
Fixes the Herald being able to drag it's mirror around. (#68021)


About The Pull Request

Fixes the Herald being able to drag it's mirror around.
Why It's Good For The Game

Requested by admins.
Changelog

cl
fix: Fixes the Herald being able to drag it's mirror around.
/cl

tgstation-server committed:
Automatic changelog generation for PR #68021 [ci skip]

san7890 committed:
Merge Conflict Markers - The Explicit Pathing, Layering, Iconing, Warning. (#68039)

* Merge Conflict Markers - The Explicit Pathing

Hey there,

This PR corrects an issue I've been having with mapmerge2 these last few months. Basically, what it does is create a base `/obj` that is given the name `---Merge Conflict Marker---`. This is fine and all, but the problem is that the base `/obj` is set to a certain plane. This does mean that sometimes, this VERY IMPORTANT marker is covered up by rocks or other objects. So, this seeks to get rid of that potential flaw, as well as do some other things.

Sometimes, when objects are rendered via GAGS or other code-means, they tend to have the same default Purple/White Sprite that any object without a valid icon_state has. This has caused me some confusion, so I have decided to create a new icon for conflict markers. This icon was designed to be as ugly as possible, while creating as much contrast as possible with the background by incorporating several colors into its design. I hope you find merge conflicts as unpleasant as I do.

I also updated mapmerge2 to have it so you can set the specified path of the object, as well as a small comment on the warnings if you do not heed it. I'm keeping the fact that mapmerge2 adds a name to the object just in case someone else really needs that. I also updated the linters to check for this path as well (even though the name and description should suffice for linting), and it should all be gravy from here.

* Adds further contrast to the DMI.

* Splits the merge conflict marker into a generic /obj

I also added a thing where if it didn't get caught by linters and it showed up on Initalize, it would error to mapping logs, spit out an error in world, and do all sorts of stuff to remind you.

* python new line

* forgot to add a tab

MrMelbert committed:
Nerfs Nettle / Deathnettle size to Normal  (#68144)

Nettles are normal sized

tgstation-server committed:
Automatic changelog generation for PR #68144 [ci skip]

vincentiusvin committed:
Adds logging for experisci completion (#68147)

Adds logging for experisci completion

tgstation-server committed:
Automatic changelog generation for PR #68147 [ci skip]

timothymtorres committed:
[NO GBP] Fix virus symptoms to happen on final stage (#68152)

* [NO GBP] Fix virus symptoms to happen on final stage

* Change disease illiterate effect to stage 5

tgstation-server committed:
Automatic changelog generation for PR #68152 [ci skip]

GuillaumePrata committed:
Updates QM RPG trim icon (#68176)

imageadd: The QM's RPG trim icon uses blue instead of brown.

tgstation-server committed:
Automatic changelog generation for PR #68176 [ci skip]

Iamgoofball committed:
Fixes the Structural Weakpoint Objective to do a normal, smaller explosion, especially on Lowpop(~20) and gives the objective a lower weight. (#67865)

tgstation-server committed:
Automatic changelog generation for PR #67865 [ci skip]

orthography committed:
Adds cameras to the Icebox Emitter room. (#68192)

Adds 2 new cameras to the Icebox Emitter room allowing the AI to access the APC malf ai moment.

Allows the AI to access the APC.

tgstation-server committed:
Automatic changelog generation for PR #68192 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

JohnFulpWillard committed:
Makes typing indicators use your bubble icon (#67872)


About The Pull Request

Typing indicators now rely on what bubble icon you have, rather than all being the default.

Most speech bubbles with Admin Combo HUD: https://cdn.discordapp.com/attachments/751307359351210014/992025794207551538/speechbubbles.mp4

A new sprite has been added for every bubble type for this to properly work, and nearly all bubble icons have been resprited by @Crumpaloo

All bubbles have been moved to the right side of the user as to not cover their HUDs, so it is now consistent with the default bubbles.
Why It's Good For The Game

This sucks:
https://user-images.githubusercontent.com/53777086/174477936-98d92891-7d2d-409c-8b9b-f92cd046edc4.mp4

It also lets us re-add one of the main perks of the Lawyer's badge, which is currently locked to not being on runechat, an option that might not exist in the future.
Changelog

cl Crumpaloo, jlsnow301, JohnFulpWillard
add: Typing indicators now use your speech bubble type, rather than all being the same, so Cyborgs no longer look like a human when typing, and Lawyers have that cool speech bubble again.
/cl

tgstation-server committed:
Automatic changelog generation for PR #67872 [ci skip]

orthography committed:
Corrects the area labels of atmospherics. (#68190)

Corrects an area of metastation that now causes the atmos alarm to enable firelocks in the atmos engine.

tgstation-server committed:
Automatic changelog generation for PR #68190 [ci skip]

stylemistake committed:
Set crossOrigin to anonymous when doing Byond.loadJs() (#68216)


About The Pull Request

Attempts to fix the "Script Error" by adding a crossOrigin attribute to the injected script node.
Changelog

cl
fix: Fixed "Script Error" in tgui and tgchat (hopefully).
/cl

tgstation-server committed:
Automatic changelog generation for PR #68216 [ci skip]

dragomagol committed:
Makes pAI screen alert actually work (#68118)

Currently, when a pAI personality is submitted, it's supposed to show an alert screen briefly with flick. This doesn't work. So I've changed it to a 5 second overlay instead, and this does work.

tgstation-server committed:
Automatic changelog generation for PR #68118 [ci skip]

SuperNovaa41 commented on code/modules/surgery/organs/autosurgeon.dm:
```diff
@@ -3,124 +3,166 @@
 	inhand_icon_state = "nothing"
 	w_class = WEIGHT_CLASS_SMALL
 
+	//how many times you can use the autosurgeon before it becomes useless
```
These all need to be proper doc comments (`///`)

Mooshimi committed:
[MDB Ignore] Power cable AI Sat update (#68105)

tgstation-server committed:
Automatic changelog generation for PR #68105 [ci skip]

CoffeeDragon16 committed:
DMDocs and updates comments

CoffeeDragon16 committed:
DMDocs and updates comments

orthography committed:
Corrects a section of icebox maintenance. (#67978)

There's a section of the security vestibule labeled vestibule maintenance but it was actually denoted as labor process, simple correction, preventing people from accidentally getting radded up the ass.

tgstation-server committed:
Automatic changelog generation for PR #67978 [ci skip]

lizardqueenlexi committed:
Made MetaStation deliveries flaps transparent. (#68202)

Replaced all opaque plastic flaps in the MetaStation deliveries area with transparent flaps.

tgstation-server committed:
Automatic changelog generation for PR #68202 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

Pepsilawn committed:
Fixes some incorrect/missing nearstation areas around Kilo's ordnance launch site (#68212)

I assume this happened when ordnance got its cold chamber for BZ production and the entire launch site had to be moved right to compensate.

tgstation-server committed:
Automatic changelog generation for PR #68212 [ci skip]

MMMiracles committed:
Tramstation: Service Wing Overhaul (again hhhhhh) (#68084)

This PR re-arranges the service departments in a way that pushes the bar/kitchen upfront and center with Hydroponics and the less important rooms taking a backstage on the lower floor.

The kitchen and bar should be front-and-center attraction for easy access and crew interaction. The fact something like aux storage was more easily accessible/visible than the bar was a weird choice ultimately. Hydroponics is still a department that gets visited by other crew, so ideally their presence on the lower floor will still suffice for giving active reason to see traffic on the lower floor.

Main Changes:

The bar is now a centralized layout that sits in front of the kitchen segment, giving a combined department area that both the chef and bartender control.

The gambling/card game tables have been moved to the side to their own 'skill game' room with dimmed lighting.

The theatre now has a more proper main stage that is easily seen from the new area entrance.

Hydroponics, custodial closet, law/courtroom and tool storages were moved downstairs as they aren't as active but still allow quick access via direct stair access from the central dock.

The surrounding maintenance was re-arranged a bit to accommodate these changes with some modular maintenance changes/removal (mainly the weird additional cave between cargo/service).

To help associate product movement between the chef and botanists now that they're on different floors, the smart fridge now has a dumbwaiter that both departments can call up/down to their room to access. Adequate protection has been added to prevent people from somehow accidentally getting crushed by said new dumbwaiter.

The service lathe has been moved downstairs with a general access room between the barman's storage room and the freezer.

The kitchen freezer is now a snow-filled room that is actually colder than the normal rooms, but not cold enough to be uninhabitable.

The one-way access helpers on sensitive maintenance doors like power hatches have been removed.

tgstation-server committed:
Automatic changelog generation for PR #68084 [ci skip]

Zonespace27 committed:
Removes the code for four removed gateways (#68038)

* gets rid of the four axed gateways's code - Wizard Academy, Challenge, Space Battle, and Wild West.

tgstation-server committed:
Automatic changelog generation for PR #68038 [ci skip]

GuillaumePrata committed:
Roller beds and bodybags are folded with right click now (#67893)

* Makes Right-click fold up roller beds and bodybags, entirely replacing MouseDrop.

tgstation-server committed:
Automatic changelog generation for PR #67893 [ci skip]

optimumtact committed:
Don't process topic calls when the ntmessenger program is qdeleted (#68206)

* Adds a check for QDELETED in NTmessenger's /Topic()

Jolly-66 committed:
[READY] [KC13] Showing "The Derelict" some love: General updates, aesthetic changes and misc (#67696)

With this PR I aim to make KC13 (TheDerelict.dmm), or Russian Station (whatever you guys call it) a tad bit more flavorful with its environment as well as somethings on the mapping side (like adding area icons!).
To preface, no, I'm not remapping anything here extensively. The general layout should be relatively the same (or should be in theory).

Halfway through naming the area icons I checked the wiki page and found out it was KC not KS, so, its KS13 internally.

Readability for turf icons are cool.
Also just making the ruin more eye appealing would be better.
General cleanup and changes will give new life to this rather.. loved? Hated? Loot pinata? Ruin.
The ruin also now starts completely depowered, like Old Station (its a Derelict, it makes no sense for it to still be powered after so long). As for some mild compensation, a few more batteries were sprinkled in to offset any issues. If there is any concern of "But they'll open the vault faster!", there were always 5 batteries that people used to make the vault SMES.
Lastly, giving it some "visual story telling" is cool, as mapping fluff goes.

I also added a subtle OOC hint that the SMES in the northern most solar room needs a terminal with the following:

SMES Jumpscare
As an aside, I aim to try and keep the feel of this ruin being "dated" while at the same time having some of our newer things. With that, certain things I'll opt out of using in favor of more "generic" structures to give KC13 that true "Its old but not really" feel and look.

tgstation-server committed:
Automatic changelog generation for PR #67696 [ci skip]

lizardqueenlexi committed:
Fixes an incorrectly-oriented disposals pipe in Tramstation Medical. (#68220)

One of the disposals junctions in Tramstation's Medical department pointed the wrong direction, which caused it to route all trash from further up the disposals network directly into the Medical breakroom. This PR replaces it with the proper junction.

tgstation-server committed:
Automatic changelog generation for PR #68220 [ci skip]

orthography committed:
Gives the BMP Asteroid away mission spawns atmos (#68179)

Hopefully gives BMP Asteroid spawns the oxygen it so desperately needs.

Creates a subset of the BMP Asteroid area spawns giving them the same initial oxygen settings as the biodomes. Hopefully this fixes that.

tgstation-server committed:
Automatic changelog generation for PR #68179 [ci skip]

Kapu1178 committed:
Improves ambience (#68041)

jlsnow301 committed:
Fixes Fission360 bluescreening the tablet (#68228)

* Fixes Fission360 app bluescreening, rendering the whole tablet useless.

tgstation-server committed:
Automatic changelog generation for PR #68228 [ci skip]

VioletN committed:
Makes the changeling tentacle power function like its description says it does (#68023)

* Makes changeling tentacle work like the description says it does, and polishes up the description a bit to make it more clear how it functions.
* replace intent with combat mode, and merges procs tentacle_stab() and tentacle_grab() together.

tgstation-server committed:
Automatic changelog generation for PR #68023 [ci skip]

Profakos committed:
Sets minimum player count for sentient disease (#68213)

tgstation-server committed:
Automatic changelog generation for PR #68213 [ci skip]

CursedBirb committed:
Qm coat now subtype of department coats like other heads coats (#68229)

Qm wintercoat has been made like into subtype like other heads coats. Qm winter coat can carry telebaton now too

tgstation-server committed:
Automatic changelog generation for PR #68229 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

orthography committed:
Wired the Atmos Engine APC (#68234)

Made a new APC placement in a previous pull (on Metastation), forgot to wire it, oopsie doopsie!!

tgstation-server committed:
Automatic changelog generation for PR #68234 [ci skip]

zxaber committed:
Fixes airlock shock indicators not being removed when the airlock is unshocked (#68112)

shocking development

tgstation-server committed:
Automatic changelog generation for PR #68112 [ci skip]

13spacemen committed:
Fixes Survival Pod Walls Not Smoothing With Their Airlocks (#68116)



Fixes #68115 (Survival Pod Walls Don't Smooth With Their Airlock)

They used to smooth, but then during the icon smoothing refactor someone forgot to add them to their smoothing_groups
Also corrected the misspelling

Tastyfish committed:
Adds unit tests for bar sign datums (#68046)

kawoppi committed:
Refactors cleaning into a component (#68156)

* Completely refactors cleaning tools to all use a single component rather than copy pasting everywhere.
* Everything that cleans (including Cleanbots and the like) all use this component. It is called on an atom-level proc, to prevent signal overuse/misuse.
* This additionally means that cleaning is much more consistent between tools, for stuff like cleaning blood off of windows and soap not being able to clean worn clothes, since it isnt a problem anymore.

tgstation-server committed:
Automatic changelog generation for PR #68156 [ci skip]

MrMelbert committed:
Fix some dashes breaking permanently (heirophant staff and ninja sword) (#68195)

* Fix dashing sometimes breaking, with additional code improvement to go along with it.

tgstation-server committed:
Automatic changelog generation for PR #68195 [ci skip]

Sucy-Manbavaran committed:
Adds Sugar to Biogenerator (#68254)

Adds sugar to the Biogenerator's usually found in botany

Toto5561 committed:
The Bacon Has Been Improved (#68255)

Updated "baconb" and "baconcookedb" sprites. Changed code to reflect the sprite changes.

tgstation-server committed:
Automatic changelog generation for PR #68255 [ci skip]

Sheits committed:
Removes a stray pixel on the base human female chest sprites (#68252)

Saves the codebase

tgstation-server committed:
Automatic changelog generation for PR #68252 [ci skip]

lizardqueenlexi committed:
Fixed multiple problems with the Tramstation deliveries system. (#68250)

Fixed multiple problems with the Tramstation deliveries system, and added several missing names.

tgstation-server committed:
Automatic changelog generation for PR #68250 [ci skip]

VladinXXV committed:
Gives tacticool turtlenecks and skirtlenecks suit sensors (#68240)

Cue people complaining for no reason. If you don't want them on just disable them. No one can force you to enable them. Trust me, every medical player ever has TRIED.

tgstation-server committed:
Automatic changelog generation for PR #68240 [ci skip]

Zonespace27 committed:
Allows the mass downloading of server logs (#68232)

* allows mass downloading of logs

* ack, indentation

* mothblock requested a change, i request that they make their name singular

* fix

tgstation-server committed:
Automatic changelog generation for PR #68232 [ci skip]

Wallemations committed:
Gives sign language users a unique speech indicator, as well as a new tonal indicator (#68231)

* don't feed the branch title monster

* timer & indicator client

* prefs & timer

tgstation-server committed:
Automatic changelog generation for PR #68231 [ci skip]

jlsnow301 committed:
orbit ui hotfix (#68224)

reeeeeeeeeeee

tgstation-server committed:
Automatic changelog generation for PR #68224 [ci skip]

san7890 committed:
Adds the Ticket #ID To AdminPM Logs (#68215)

Hey there,

Statbus has been broken for 2 months+, and what I've been doing when I need to reference a ticket is to go into the raw-logs and parse it for the involved CKey. However, you tend to get a lot of noise searching for the signal when you filter in Notepad++, so I decided today would be a good day to have the Ticket #ID in all associated AdminPMs (which are already logged in ADMINPRIVATE).

tgstation-server committed:
Automatic changelog generation for PR #68215 [ci skip]

QuacksQ committed:
Paper walls desc spelling fix (#68210)

Fixed a weird description on paper walls

tgstation-server committed:
Automatic changelog generation for PR #68210 [ci skip]

optimumtact committed:
Don't assume computers and applications still exist when broadcasting (#68207)

subspace messages

optimumtact committed:
Custom reagent holders no longer blow through the overlay cap (#68205)

carshalash committed:
Removes upload boards from steal objectives. (#68204)

* Update objective_items.dm

* I always forget one thing.

tgstation-server committed:
Automatic changelog generation for PR #68204 [ci skip]

GoldenAlpharex committed:
Fixes trapdoor runtime when they're destroyed when not linked to a remote (#68189)

Fixes trapdoor runtime when they're destroyed

tgstation-server committed:
Automatic changelog generation for PR #68189 [ci skip]

sergeirocks100 committed:
Gives the library computer/old computer directional sprites. (#68187)

initial commit

tgstation-server committed:
Automatic changelog generation for PR #68187 [ci skip]

LemonInTheDark committed:
Mild plane tweaks, thermomachine dropshadow (#68185)

Adds the area plane to the game and colorblind plane master controllers
Mostly because it's actually used now, for weather, and should be
blurred/recolored

Sets the thermomachine on the game plane
I want it to have a dropshadow, and right now it's on the floor plane,
just from atmospherics, which I think is stupid

tgstation-server committed:
Automatic changelog generation for PR #68185 [ci skip]

Fikou committed:
ethereals no longer revive after being changeling drained (#68184)

* ethereals no longer revive after being changeling drained

* Drain() on absorb proc is applied after mob death. This means the signal will run first and our code won't matter, so let's run it on crystallize instead!

tgstation-server committed:
Automatic changelog generation for PR #68184 [ci skip]

vincentiusvin committed:
Buffs heat, pressure limit of pumps and scrubbers. Buffs can release pressure. (#68175)

* Heat and pressure buff

* Canister buff

tgstation-server committed:
Automatic changelog generation for PR #68175 [ci skip]

unit0016 committed:
Revisiting The Goliath: Or, that time I dripped out the SBC Starfury just because (#68126)

Drips the SHIT out of the SBC Starfury while not completely overhauling it. Touches everything NOT in engineering or southward (because I love how scuffed that part is and refuse to touch it on principle) - Also converts one map varedit into a real boy subtype, and moves tiny fans to their own file.

Mandatory disclosure on the gameplay changes:
Fighters 1 and 3 are now NOT in the hangar, and are now attached to the formerly unused gunnery rooms.
Cryo now works. Yeah. I know.
You can actually open the anesthetic closet now.
Everyone now shares three spawners. This doesn't reduce the amount of people who can play when this rolls, as I've adjusted var/uses in accordance: it just reduces clutter.
A few of the horizontal double airlocks have been compacted into glass_large airlocks.
The bar windows now actually have grilles like they were meant to.
Four turbines have shown up. They aren't functional*, they just look like gunnery and conveniently fit in the spots. I'm sure this is space OSHA compliant.
The map is ever so slightly smaller, vertically. This should distance us from an edge case where somehow all space levels are too cluttered for this to spawn properly, for the time being.

*Technically there's nothing stopping you from using them besides the amount of time it'd take for the operatives to kick your ass

This map was originally designed wayyy back before we even had the computer sprites we have now, (#27760 if you want to see SOUL) and it shows. While it will never have it's SM again, we can at least make the thing much nicer to look at.

tgstation-server committed:
Automatic changelog generation for PR #68126 [ci skip]

13spacemen committed:
Experimental Syndicate Teleporter (#68029)

* adds the syndicate teleporter to the game
available at uplink for 8 tc
lets user teleport 4-8 tiles forward, recharges randomly, can gib if no safe emergency location or EMPd/wearing bag of holding

tgstation-server committed:
Automatic changelog generation for PR #68029 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

azrael-x86 committed:
Tsu's Brand Spanking New Storage: or, How I Learned To Pass Github Copilot As My Own Code (#67478)

Currently, storage works as a subtype of /datum/component, utilizing GetComponent() and signals to operate. While this is a pretty good idea in theory, the execution was pretty trash, and we end up with alot of GetComponent() snowflake code (something that shouldn't even need to be used frankly), and a heaping load of scattered procs that lead into one another, and procs that don't get utilized properly.

Instead, this PR adds atom_storage and proc/create_storage(. . .) to every atom, allowing for the possibility of storage on quite frankly anything. Not only does this entirely remove the need for signals, but it heavily squashes down the number of needed procs in total (removing snowflake signal procs that just lead to one another), reducing overall proc overhead and improving performance.

tgstation-server committed:
Automatic changelog generation for PR #67478 [ci skip]

optimumtact committed:
New player job selection fixes pack (#68208)

* New player job selection fixes pack

If no jobs are available when random jobbing dont just runtime instead
pop a little tgui alert letting them know they are shit out of luck

Also makes the job debug actually print out the job titles instead of
just printing a . like some kind of insane person

Also also removes some syntax left over from a search and replace

* Update code/modules/mob/dead/new_player/new_player.dm

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>

* Update code/modules/mob/dead/new_player/new_player.dm

Co-authored-by: Tom <8881105+tf-4@users.noreply.github.com>

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>
Co-authored-by: Tom <8881105+tf-4@users.noreply.github.com>

cacogen committed:
Makes long numeric station name rarer (#68218)

Makes long-ass station name rarer

Also makes it use normal prefixes rather than the generic "Space Station", which saps all the joy out of life

tgstation-server committed:
Automatic changelog generation for PR #68218 [ci skip]

MrMelbert committed:
Fixes Bolts of Necropotence  (#68260)

Fixes necropotence due to a copy and paste error.

tgstation-server committed:
Automatic changelog generation for PR #68260 [ci skip]

jjpark-kb committed:
trailer modification and safe qdel (#68095)

* trailer modification and safe qdel

* suggested change

tgstation-server committed:
Automatic changelog generation for PR #68095 [ci skip]

timothymtorres committed:
Fix deafness virus symptom to be permanent if resistance threshold is met (#68080)

* Fix deafness virus symptom to be permanent

* Refactor deafness disease trait

Co-authored-by: ATH1909 <42606352+ATH1909@users.noreply.github.com>

* Add disease trait on stage 5

Co-authored-by: ATH1909 <42606352+ATH1909@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #68080 [ci skip]

timothymtorres committed:
Fix simple mob deaths causing deadchat notifications (#68155)

* Fix simple mob death message deadchat notification

* Add TRAIT_DEATH_GHOST_ALERT to traits

* Add TRAIT_DEATH_GHOST_ALERT to global trait list

* Add TRAIT_ALERT_GHOSTS_ON_DEATH to space dragons

* Add TRAIT_ALERT_GHOSTS_ON_DEATH to brood spiders

* Add TRAIT_ALERT_GHOSTS_ON_DEATH to elite lavaland mobs

* Add TRAIT_ALERT_GHOSTS_ON_DEATH to revenant

* Add TRAIT_ALERT_GHOSTS_ON_DEATH to morph

* Fix trait name

* Fix trait name

* Fix syntax error for trait

* Add trait check for death notification

* Add more context to trait use

tgstation-server committed:
Automatic changelog generation for PR #68155 [ci skip]

LemonInTheDark committed:
Fixes a compile error caused by #68080 (0b7e5429273901c230bc5d8bf9864569232a1fa2) (#68282)

Local man forgets to test, misnames a var
I cleaned up M here too because it pissed me off

Oh, also the unit tests didn't properly run on the last commit, which
fooled the merging maintainer into assuming it passed.

WTF happen there

MrMelbert committed:
ICEBOX arrivals: Reworks EVA to make more sense for a snow planet, featuring cool unique coats. Also updates the style and layout of various rooms along the arrivals hall. (#68099)

Iceboxstation's EVA has been remapped.
Within the EVA room now contains multiple kits of unique equipment that better help people who wish to brave the wilderness.

EVA suits have been replaced with EVA coats - these coats are a bit closer to EVA suits in their stats and usage.

They're bulky sized.
They apply a bit of slowdown while worn.
They can carry a full sized oxygen tank in their suit storage slot.
They have armor (about half the armor of EVA suits), and are inflammable
Additionally, it also comes stocked with EVA boots, which are a subtype of the miner's snowshoes that slow you down while worn.

The command section of EVA contains an old EVA suit if one is needed for some odd reason.

(But what about xeno protection ahhhh) (It still has welding helmets I guess)

EVA picture within (click the triangle)
To not stick out like a sore thumb, the arrivals hallway in general has been dolled up, updated, and modernized.
Here's the gateway and vault.
Tool storage and garden.
Art storage and emergency storage. (Bonus: The commissary has a maint airlock again.)
Other miscellaneous changes:

Cleaned up the pipenet that goes around arrivals and into cargo.
Same with cables, but to a lesser extent.
One maint room has been added.
Port maint and lesser port maint has their area definitions swapped around, and less port maint gained a little size.
Intercom coverage around the place has been improved.
Some greenery was added to the snow pockets around the bridge.
Vault has a small moat on the right side of it that goes to the lower levels. Watch out.

Icebox's EVA was a bit uh... well it stuck out for what it was used for.
EVA suits were entirely outclassed by winter coats, and magboots I think didn't even do anything to being slipped on ice.
By adding some unique coats, ideally people have more of a reason to need to visit EVA instead of just grabbing a coat from dorms / the locker room.

On top of it, the entire hallway had some very dated design and lacked decalling and style. Updating it makes it a lot more visually appealing and less drab.

tgstation-server committed:
Automatic changelog generation for PR #68099 [ci skip]

orthography committed:
Changes a Tramstation "lil pump" to a regular one. (#68267)

Changes a "lil pump" to a normal pump to prevent the space-time anomalies San warns us about.

tgstation-server committed:
Automatic changelog generation for PR #68267 [ci skip]

orthography committed:
Corrects a smart pipe improperly connecting on Tramstation. (#68268)

Moves the pipes around to prevent smart pipes from improperly connecting without having to change the colors.

tgstation-server committed:
Automatic changelog generation for PR #68268 [ci skip]

Commit:
dehardcodes admin grantable objectives list (#68245)

* Update objective.dm

* Update objective.dm

* Update objective.dm

* Update code/game/gamemodes/objective.dm

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

* Update objective.dm

Co-authored-by: John Willard <53777086+JohnFulpWillard@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #68245 [ci skip]

StrangeWeirdKitten committed:
Fixes Bubblegum (#68266)

* Update bubblegum.dm

* Update code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm

Co-authored-by: ShizCalev <ShizCalev@users.noreply.github.com>

* Ok

Ok

Co-authored-by: ShizCalev <ShizCalev@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #68266 [ci skip]

YakumoChen committed:
Makes GPS units autolathe banned (#68100)

* Bans GPS' from being added to Autolathes' Imported section.

tgstation-server committed:
Automatic changelog generation for PR #68100 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

san7890 committed:
Fixes Gulag Shuttle Access On Tramstation (#68221)

* Fixes Gulag Shuttle Access On Tramstation

Hey there,

The prisoners actually enter on the LEFT, not the right, so these accesses were not actually set properly. Let's fix that up right now.

* Revert "Fixes Gulag Shuttle Access On Tramstation"

This reverts commit dd1b1c25ad5351cdba515c2f0d77494f409f7ed2.

* ok hard reset and redo

tgstation-server committed:
Automatic changelog generation for PR #68221 [ci skip]

MrMelbert committed:
Fixes Tonic Water causing dizziness (#68276)

Fixes tonic water causing dizziness

tgstation-server committed:
Automatic changelog generation for PR #68276 [ci skip]

san7890 committed:
Fixes TramStation's Xenobiology Disposals (#68286)

tgstation-server committed:
Automatic changelog generation for PR #68286 [ci skip]

san7890 committed:
[s] Patches potential runtimes in AdminPM (#68324)


About The Pull Request

Hey there,

6e29689 seemed to work alright on my end, but apparently there are cases where current_ticket.id = NULL. This would cause runtime trace errors that would start to list out the first few words of most adminhelps. Fuck.

You probably could use recipient.current_ticket.id on the "else" portion but I woke up to this and I'm not 100% certain if that will null out at this time. Might also be apt to revert #68215 entirely?
Why It's Good For The Game

Those shouldn't show up in runtimes!
Changelog

Nothing particularly player facing.

LemonInTheDark committed:
Refactors mothweek check (#68280)

* Refactors mothweek check

Mothweek HAS SIDE EFFECTS
Checking it on a day that causes a week offset will currently cause that
week offset to persist
That's dumb.

Also it's not very expansive, only covers a slim set of possibilities.

Instead, lets build something to generate all passing days over a period
of time, maybe 3 months out of 2 years.
Then we'll crosscheck that against some predecided "ok" dates

If either list disagrees with each other, we'll fail. That way we can't
miss an edgecase. or have issues with side effects

I like this pattern.

actions-user committed:
Automatic changelog compile [ci skip]

azrael-x86 committed:
[NO GBP] changes box storage limit to 2 instead of 4 (#68298)

fun ruined

tgstation-server committed:
Automatic changelog generation for PR #68298 [ci skip]

13spacemen committed:
Shortens Hub Entry Descriptors (#68186)

Timberpoes committed:
Removes temporary debugging code that can spam admin chat. (#68321)

Feex

tgstation-server committed:
Automatic changelog generation for PR #68321 [ci skip]

san7890 committed:
Shady Backroom Deals - Bartender's Area Definition (#68311)

add: bartender's cubby area is now known as the "backroom"

tgstation-server committed:
Automatic changelog generation for PR #68311 [ci skip]

LemonInTheDark committed:
Converts inappropriate update_overlays -> update_appearance (#68315)

IT DOESN'T DO ANYTHING
update_overlays is a helper proc, it returns a list of overlays to
add/track.
It doesn't update anything on its own.

Kapu1178 committed:
Ambience tweaks (#68264)

* cleans up code and adjusts volume

* Removes can_hear() check

* volume

* review

tgstation-server committed:
Automatic changelog generation for PR #68264 [ci skip]

LemonInTheDark committed:
Prevents overtime in the economy SS (#68078)

* Prevents overtime in the economy SS

Ok so like, we were doing a LOT of work all at once without yielding,
which is fine cause like, it's only fired once every 5 minutes, but it's
still kinda dumb

So I've cleaned up the subsystem to make it support an ssair style of
yielding, alongside an optimization to vending machine z level checks
that I suspect was taking a lot of cpu.

Remind me to refactor z level lookup someday bestie

Anyway yeah now we won't see strangely high overtime for ssecon anymore

* vending gang (renames a variable to not just be v

* It is done

azrael-x86 committed:
[NO GBP] properly updates the parents icon when an item is added to storage (#68305)

tgstation-server committed:
Automatic changelog generation for PR #68305 [ci skip]

RimiNosha committed:
Un-headache-ify Lavaland's Ambiance! (#68323)

tgstation-server committed:
Automatic changelog generation for PR #68323 [ci skip]

Couls committed:
Fixes for AI detector (#68301)

Fixes multitool detection

Also modifies the uplink description to reflect the changes made to the multitool and makes the HUD mode into toggleable detection mode, and removes the unused hud code

tgstation-server committed:
Automatic changelog generation for PR #68301 [ci skip]

azrael-x86 committed:
[NO GBP] fixes issue where observers couldn't open storage items (#68299)

* fun ruined

* shit

* woops

tgstation-server committed:
Automatic changelog generation for PR #68299 [ci skip]

zxaber committed:
Combat mech research nodes now require mech-themed experiments rather than bomb experiments. (#67756)

* Mechs require a new Mech scanning experiment centered around damaging/destroying mechs, instead of being behind weaponry, which requires ordnance research.

tgstation-server committed:
Automatic changelog generation for PR #67756 [ci skip]

Salex08 committed:
Makes silver spawned food actually disgusting (#68319)

Fixes silver spawned food not tasting disgusting by sending a signal to the Edible component to make it disgusting.

tgstation-server committed:
Automatic changelog generation for PR #68319 [ci skip]

san7890 committed:
Adds an Air Alarm to TramStation Bar Lounge (#68308)

tgstation-server committed:
Automatic changelog generation for PR #68308 [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

Spookuni committed:
Gives Delta genetics a "back-desk" to encourage interaction (#68340)

Gives Delta genetics a basic desk leading into the science maint area to make interacting with them a little bit easier.

Having open desks makes it less likely that people succumb to the siren song of being a job-completing gremlin who never talks to anyone else, interacts with them or uses their job content to benefit the station as a whole. A fully public front desk would be preferable for this, but it would require a major rework of the science department that is beyond my meagre mapping abilities. Letting geneticists more easily interact with passing assistants/secoffs/parameds/engineers/anyone with maint access will hopefully help a little in the fight against job-gremlinism.

tgstation-server committed:
Automatic changelog generation for PR #68340 [ci skip]

unit0016 committed:
Icebox Murals: AI Upload and RD Office (#68341)

Redoes the Icebox upload slightly. Adds murals to the upload and RD's office. Also compresses the tile decals on the bridge.

Murals are fun ways to implement art into spaces, whether they commune something or are just there to look nice. It's weird we don't have any in-game already.

tgstation-server committed:
Automatic changelog generation for PR #68341 [ci skip]

jlsnow301 committed:
Fixes admin.fun list (#68336)


About The Pull Request

Many of the items in the list of admin fun were not actually categorized as Admin.Fun (and thus not appearing), and some admin fun procs were not even listed.
Why It's Good For The Game

Basically, more admin fun procs are available that were already in code but not listed
Changelog

cl
admin: Fixed many "admin.fun" listings that were incorrect or missing.
/cl

tgstation-server committed:
Automatic changelog generation for PR #68336 [ci skip]

13spacemen committed:
Removes useless colon in hub entry (#68347)

Removes unnecessary colon in hub entry.

ReinaCoder committed:
Resprites the safe and floor safe (#68284)

imageadd: The safe and floor safes have been resprited

tgstation-server committed:
Automatic changelog generation for PR #68284 [ci skip]

EOBGames committed:
Poster Pack 2 - Beyond Posters (#68235)

Adds a new selection of posters, poster-like objects, and wall decorations.

New entirely with this PR, shipping containers! These sprites were adapted from a base from TGMC, with brands and designs more fitting to our universe here on TG

Another new addition, flags! These draw upon Common Core lore, representing several major factions from the lorebase as a new mapping tool

Finally, there are two interactive wall objects- wall clocks, and calendars:

The wall clock shows both the "real time", represented as the Earth time, or Treaty Coordinated Standard Time, and the station time, represented as the local time, or Coalition Standard Time

The calendar gives the in-universe date, as well as any active holidays

Why It's Good For The Game

This adds a lot of fun aesthetic additions that I think will benefit our mapping team in creating new spaces and breathing new life into older ones. Additionally, it helps make our Common Core lore a bit more tangible to the playerbase, bringing it away from being "just on a wiki" and into the game proper.

A whole host of new posters have just become available.
For the architects among you: a number of new aesthetic options are available for mapping with. Let your creativity free!

tgstation-server committed:
Automatic changelog generation for PR #68235 [ci skip]

Thunder12345 committed:
Mining Outpost Overhaul: Island Living Edition (#67796)

Replaced the old mining outpost with a new base consisting of separate buildings on islands in the middle of lava. Added a few new areas and decorations to fill out the outpost. Changed the underfloor accessibility of lava so cables and pipes don't mysteriously disappear as they cross catwalks over lava.

The current mining station is fairly boring and dated. In addition, the general layout of the outpost was bland and uninspired, about as exciting as any other hallway. This replacement offers a new and unique take with an interesting layout and new dangers to be aware of.

The bridges between buildings are exposed to the weather, meaning you could find yourself forced to choose between waiting out a storm in an isolated building with a potentially unfriendly roommate, or running through the storm to get back to a shuttle.

Power and communications are also more vulnerable in the new base, placed in a building on the mainland where ne'er-do-wells could gain access unnoticed and sabotage them, making the mining outpost less of a safe space with little risk of anything going seriously wrong. You don't get access to these just for having mining access, better call the relevant portion of engineering to help you.

Gulag prisoners also get a few new options to keep security on their toes, with a few different options for escape routes that encourage the warden to keep an eye on their charges instead of ignoring the gulag while waiting for the prisoners to do their sentences or kill themselves.

tgstation-server committed:
Automatic changelog generation for PR #67796 [ci skip]

ShizCalev committed:
Fixes some door shocking issues (#68278)

* Fixes some door shocking issues

tgstation-server committed:
Automatic changelog generation for PR #68278 [ci skip]

orthography committed:
Asimov++ made editable identically to Asimov (#68295)

Co-authored-by: Mothblocks <35135081+Mothblocks@users.noreply.github.com>

tgstation-server committed:
Automatic changelog generation for PR #68295 [ci skip]

Thunder12345 committed:
Revert "Fixes admin.fun list" (#68369)

Revert "Fixes admin.fun list (#68336)"

This reverts commit c17acd07e64a72453cd85b99cd3e1d9a080b78fc.

CoffeeDragon16 committed:
Golems now have tongues again, except cardboard golems (#68168)

* vocal cord slot

* puts vocal chords back

* reverts changes i couldnt figure out how to make mutant_vocalcords work

* cleanup

* woops

* Makes cardboard golems have no tongues

* adds no tongue to the greet

tgstation-server committed:
Automatic changelog generation for PR #68168 [ci skip]

CoffeeDragon16 committed:
Merge remote-tracking branch 'origin/autosurgeonfix' into autosurgeonfix

CoffeeDragon16 committed:
Revert "Merge remote-tracking branch 'origin/autosurgeonfix' into autosurgeonfix"

This reverts commit 3782e1a3da1bd47fb57623875f4350896bfe8667.

CoffeeDragon16 commented:
i bonked the branch hard, gonna re-open it on a new one
